include/uk/plat/mte.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2022, Michalis Pappas <mpappas@fastmail.fm>.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef __UK_MTE_H__
+#define __UK_MTE_H__
+
+#include <stddef.h>
+#include <uk/asm/mte.h>
+#include <uk/essentials.h>
+
+#define MTE_ALIGN_SIZE(sz) ALIGN_UP(sz, MTE_TAG_GRANULE)
+
+void *mte_tag_contig(void *ptr, size_t size);
```

Comment:
> If you use __sz you could avoid including stddef.h in a UK global header file.

Ok

> Since @craciunoiuc asked. I also stumbled over contig and first thought it was a typo and should be config. Question is: Is there a case where you would tag a non-contiguous buffer?

Funny, to me it's completely obvious, but maybe it's because it's frequently used in academia in the UK. Anyway, since it's confusing for others I'll rename to `mte_tag_region()`. The intention was to be forwards compatible, in case for example we export a function to tag a single address.

Reply:
`Question is: Is there a case where you would tag a non-contiguous buffer?`

Exact same thing I asked myself too. `contig` suggests there is a case where there will also be a `non_contig`, but maybe I'm just too pedantic

Reply:
> Funny, to me it's completely obvious, but maybe it's because it's frequently used in academia in the UK. Anyway, since it's confusing for others I'll rename to mte_tag_region(). The intention was to be forwards compatible, in case for example we export a function to tag a single address.

I like that solution and we can easily add a single address version if needed.

Reply:
Ok, that was updated in the last iteration, we can resolve this one now 👍🏼